explorer/src/components/instruction/bonfida-bot/types.ts

```diff
@@ -0,0 +1,429 @@
+/* eslint-disable @typescript-eslint/no-redeclare */
+import { PublicKey, TransactionInstruction } from "@solana/web3.js";
+import {
+  enums,
+  number,
+  type,
+  Infer,
+  create,
+  array,
+  string,
+  optional,
+} from "superstruct";
+import {
+  Instruction,
+  decodeInstruction,
+  BONFIDABOT_PROGRAM_ID,
+} from "bonfida-bot";
+
+export const SERUM_DECODED_MAX = 6;
+
+export type Side = Infer<typeof Side>;
+export const Side = enums([0, 1]);
+
+export type OrderType = Infer<typeof OrderType>;
+export const OrderType = enums([0, 1, 2]);
+
+export type InitializeBot = {
+  systemProgramAccount: PublicKey;
+  rentSysvarAccount: PublicKey;
+  splTokenProgramAccount: PublicKey;
+  poolAccount: PublicKey;
+  mintAccount: PublicKey;
+  payerAccount: PublicKey;
+
+  poolSeed: string;
+  maxNumberOfAsset: number | undefined | null;
+  numberOfMarkets: number;
+  programId: PublicKey;
+};
+
+export const InitializeBotDecode = type({
+  poolSeed: string(),
+  maxNumberOfAsset: optional(number()),
+  numberOfMarkets: number(),
+});
+
+export const decodeInitializeBot = (
+  ix: TransactionInstruction
+): InitializeBot => {
+  const decoded = create(
+    decodeInstruction(ix.data, Instruction.Init),
+    InitializeBotDecode
+  );
+  const initializeBot: InitializeBot = {
+    systemProgramAccount: ix.keys[0].pubkey,
+    rentSysvarAccount: ix.keys[1].pubkey,
+    splTokenProgramAccount: ix.keys[2].pubkey,
+    poolAccount: ix.keys[3].pubkey,
+    mintAccount: ix.keys[4].pubkey,
+    payerAccount: ix.keys[5].pubkey,
+    poolSeed: decoded.poolSeed,
+    maxNumberOfAsset: decoded.maxNumberOfAsset,
+    numberOfMarkets: decoded.numberOfMarkets,
+    programId: ix.programId,
+  };
+
+  return initializeBot;
+};
+
+export type CreateBot = {
+  splTokenProgramId: PublicKey;
+  clockSysvarKey: PublicKey;
+  mintKey: PublicKey;
+  poolKey: PublicKey;
+  poolSeed: string;
+  targetPoolTokenKey: PublicKey;
+  serumProgramId: PublicKey;
+  signalProviderKey: PublicKey;
+  depositAmounts: number[] | undefined | null;
+  markets: string[] | undefined | null;
+  feeCollectionPeriod: number;
+  feeRatio: number;
+  programId: PublicKey;
+};
+
+export const CreateBotDecode = type({
+  poolSeed: string(),
+  feeCollectionPeriod: number(),
+  feeRatio: number(),
+  depositAmounts: array(number()),
+  markets: array(string()),
```

Comment:
This `CreateBotDecode` decoder is failing on the markets array. Instead of an array of strings being returned from `decodeInstruction(ix.data, Instruction.Create)`, an array of `PublicKey`s are being returned.

One thing you could do here is write a custom `PublicKey` coercer that returns a string:
```
const PublicKeyToString = coerce(
  string(),
  any(), // usually this would be instance(PublicKey), but there appears to be a compatibility issue between Explorer web3 and bonfida-bot web3 PublicKeys
  value => value.toBase58()
);
```

Then your decoder would look like:
```
export const CreateBotDecode = type({
  poolSeed: string(),
  feeCollectionPeriod: number(),
  feeRatio: number(),
  depositAmounts: array(number()),
  markets: array(PublicKeyToString),
});
```